cmd/internal/flowmode/cmd_run.go

```diff
@@ -262,6 +271,7 @@ func (fr *flowRun) Run(configPath string) error {
 			clusterService,
 			otelService,
 			labelService,
+			remoteCfgService,
```

Comment:
the service is created and will run for all agents? That would mean that we create and run a controller without any config, can this be a problem?

Reply:
I don't think there would be a problem here and toggling the service on and off dynamically would increase the complexity of the feature.

Do you have any specific concerns in mind?

Reply:
If a user doesn't configure it via the corresponding river block, then there is no use to run the service because there is no URL to retrieve the config from, right? If yes then we can just create the service in the loader when we populate the nodes and only if there is a block for it.
What happens currently when you don't use the service but it still runs? Would it pick up a cached config if it finds one? It might also log things and spawn go routines (like the worker pool) which may bring confusions for the users

Reply:
Sorry I was writing the comment above before refreshing the page so i did not see the last comment

Reply:
We've caught up offline about a potential way to conditionally run pipelines. Since there isn't a pressing need for that (the overhead is just an extra running goroutine), and that it'd work differently for other services, we might revisit in the future but decided to not do anything for now.